packages/backend/src/services/CatalogService/CatalogService.ts

```diff
@@ -98,10 +102,32 @@ export class CatalogService<
         }, []);
     }
 
-    private static async getCatalogTables(
+    private static isExploreFiltered(
+        tablesConfiguration: TablesConfiguration,
+        explore: Pick<SummaryExplore, 'tags' | 'name'>,
+    ) {
+        // Optimization tip:
+        // Perhaps we should filter these in catalong on `index` rather than in real time (requires dbt refresh on table config changes)
```

Comment:
You mean doing it that way **will** require dbt refresh on table config changes? Is that not true anyway? Or I guess tag changes don't require dbt refresh?

Reply:
> tag changes don't require dbt refresh
This, tag changes don't require dbt refresh, but if we filter them out on index, it will be required. 

We can't do this anyway if we are going to `include` them on an advance filter.